### ISSUE TITLE
Prisma generate doens't work in Docker

### ISSUE THREAD

**Issue by faejr** [oven-sh/bun#3827]
### What version of Bun is running?

0.7

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

`prisma generate` appears to do nothing when run with bunx in docker. It tries to spawn ldconfig but I assume it fails (missing implementation?)

1. Setup a simple a project`bun init`
2. Install prisma `bun add prisma`
3. Initialize prisma `bunx prisma init`
4. Create a simple Dockerfile
```Dockerfile
FROM oven/bun:0.7

ENV DEBUG="*"

COPY . .

RUN bun install

RUN bunx prisma generate
```
5. Build image `docker build --no-cache . -t bun-issue --progress plain`

### What is the expected behavior?

That the prisma client is correctly generated

### What do you see instead?

```
#6 [4/4] RUN bunx prisma generate
#6 sha256:1987cf59e4281a608b0fae3fec4c7e0f1a98336c114e0a6300b9003e8a47b68b
#6 0.314 [0.05ms] ".env"
#6 0.742 2023-07-26T22:40:10.559Z prisma:engines  binaries to download libquery-engine, schema-engine
#6 0.744 2023-07-26T22:40:10.562Z prisma:get-platform  Found distro info:
#6 0.744 {
#6 0.744   "targetDistro": "debian",
#6 0.744   "familyDistro": "debian",
#6 0.744   "originalDistro": "debian"
#6 0.744 }
#6 0.744 2023-07-26T22:40:10.562Z prisma:get-platform  Trying platform-specific paths for "debian" (and "ubuntu")
#6 0.744 2023-07-26T22:40:10.562Z prisma:get-platform  Falling back to "ldconfig" and other generic paths
#6 0.745 spawn {
#6 0.745   cwd: null,
#6 0.745   env: null,
#6 0.745   shell: true,
#6 0.745   signal: undefined,
#6 0.745   file: "sh",
#6 0.745   args: [ "sh", "-c", "ldconfig -p | sed \"s/.*=>s*//\" | sed \"s|.*/||\" | grep libssl | sort | grep -v \"libssl.so.0\""
#6 0.745   ],
#6 0.745   envPairs: {
#6 0.745     HOME: "/root",
#6 0.745     DATABASE_URL: "postgresql://johndoe:randompassword@localhost:5432/mydb?schema=public",
#6 0.745     npm_execpath: "/usr/local/bin/bunx",
#6 0.745     npm_config_local_prefix: "/home/bun/app/",
#6 0.745     npm_package_name: "db",
#6 0.745     PATH: "/tmp/prisma@latest--bunx/node_modules/.bin:/tmp/bun-node:/home/bun/app/node_modules/.bin:/home/bun/app/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
#6 0.745     npm_package_json: "/home/bun/app/package.json",
#6 0.745     npm_config_user_agent: "bun/0.7.0 npm/? node/v18.15.0 linux arm64",
#6 0.745     DEBUG: "*",
#6 0.745     npm_node_execpath: "/tmp/bun-node/node",
#6 0.745     PWD: "/home/bun/app",
#6 0.745     NODE: "/tmp/bun-node/node",
#6 0.745     TZ: undefined
#6 0.745   }
#6 0.745 }
#6 DONE 0.8s
```

**Comment by Jarred-Sumner** [comment]
linux/arm64 is a little suspicious

hm

**Comment by faejr** [comment]
Well seeing as I'm running an m1 macbook pro I think it makes sense that docker also runs linux/arm64

**Comment by faejr** [comment]
Diving a bit deeper into this I decided to test the `child_process.exec` function using bun in the official docker image and I never seem to get any result, instead it just logs similar output as seen above by prisma.

**Comment by BehnH** [comment]
Looks like this happens even when using Ubuntu as a base image, with Bun manually installed

**Comment by faejr** [comment]
yeah i've tried that as well. the only way it works is if you install node as well, either into the bun image or whatever other image you have, since bun isn't complete enough to run prisma on its own

**Comment by maietta** [comment]
> 

You REALLY ought to be clear on that from the beginning.  You are using very odd hardware and there's a lot of work to be done to get software working on the M1 and M2 chip instructions from upstream vendors. From now on when you open an issue, could you PLEASE reference the odd hardware setup?

Had I not found this nugget of info, I would have immediately abandoned Bun due to what appeared to be a general issue with not being able to use Prisma generate.

**Comment by halfmatthalfcat** [comment]
> > 
> 
> You REALLY ought to be clear on that from the beginning. You are using very odd hardware and there's a lot of work to be done to get software working on the M1 and M2 chip instructions from upstream vendors. From now on when you open an issue, could you PLEASE reference the odd hardware setup?
> 
> Had I not found this nugget of info, I would have immediately abandoned Bun due to what appeared to be a general issue with not being able to use Prisma generate.

I'm on an Intel chip and am running into the same issue, so it may be as general as it seems. Mx chips have been also out for years now, so I wouldn't call it odd.

**Comment by henriknh** [comment]
Hey! I had a problem that the `bunx prisma generate` executed with exit code 0 thus OK but no console output nor file generation happened. The problem seems to be that that for now prisma still need access to NodeJS. See note here https://bun.sh/guides/ecosystem/prisma. After changing the image of my Dockerfile from `oven/bun` to `imbios/bun-node:20-alpine`(https://github.com/ImBIOS/bun-node) it started working.

**Comment by anhtuank7c** [comment]
OMG, I have the same issue. This is my docker file.

```
# Dockerfile
FROM oven/bun

WORKDIR /app

COPY package.json .
COPY bun.lockb .
COPY prisma prisma
COPY src src
COPY .env .env
COPY tsconfig.json .

RUN bun install
RUN bunx prisma generate
RUN bunx prisma migrate

ENV NODE_ENV production
CMD ["bun", "serve"]

EXPOSE 3000
```
